### PR TITLE
Fix hash verification for [tool.poetry.group.dev.dependencies]

### LINKED ISSUE #185: Groups support in Poetry lock files
**Is your feature request related to a problem? Please describe.**

Poetry introduced dependency groups feature that will need few adjustments in the source code to work properly:

https://python-poetry.org/blog/announcing-poetry-1.2.0a2/#dependency-groups

See also ``A note about the dev-dependencies section`` part for dev-dependencies handling. It might be also needed to rethink how we integrate with dependency groups when mapping internally to Pipfile.lock (we will probably not support them, but `dev` group can be mapped to dev-dependencies implicitly).

### PR DESCRIPTION
**Pull request by frenzymadness**
## Related Issues and Dependencies

Fixes #185 

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: tests/data/verify/poetry_group/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+name = "micropipenv-test"
+version = "0.1.0"
+description = ""
+authors = ["Fridolin Pokorny <fridolin@redhat.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.7"
+daiquiri = "2.0.0"
+
+[tool.poetry.group.dev.dependencies]
```
Comment (shifqu) [inline tests/data/verify/poetry_group/pyproject.toml:11]: Adding another group here would be a good test. Poetry 1.2 allows arbitrary names for groups.

So I can use the following command: `poetry add --group extra black`, which in turn would create a section "tool.poetry.group.extra.dependencies"

File: micropipenv.py [hunk 2]
```diff
@@ -302,6 +302,15 @@ def _compute_poetry_hash(pyproject):  # type: (MutableMapping[str, Any]) -> str
     relevant_keys = ["dependencies", "dev-dependencies", "source", "extras"]
     relevant_content = {key: poetry_data.get(key) for key in relevant_keys}
 
+    # relevant_keys are the original one and they should always be in
+    # the relevant_content even their value is None.
+    # group is a new key since poetry 1.2 and we should include it only
+    # if pyproject.toml contains it. Including it always would break
+    # backward compatibility.
+    # See: https://github.com/python-poetry/poetry/blob/4a07b5e0243bb8879dd6725cb901d9fa0f6eb182/src/poetry/packages/locker.py#L278-L293
+    if "group" in poetry_data:
```
Comment (shifqu) [inline micropipenv.py:311]: With the given fact that group might include arbitrary keys with dependencies (eg extra, dev, myspecialgroup, ...), does this code handle these cases?

**Comment by frenzymadness** [comment]
Thanks for your comments @shifqu . I've added the extra group to the test. If the CI approves this, I'm gonna merge it and release a new version.

**Pull request merged**